[Firehose] Add warning for upcoming default data stream change (#3359)
elastic/observability-docs: docs/en/aws-firehose/aws-firehose-setup.asciidoc

Find the *AWS* integration by searching or browsing the catalog.

[role="screenshot"]
image::images/aws-integrations-page.png[Integrations catalogue with the "AWS" integration highlighted]
 
Navigate to the *Settings* tab and click *Install AWS assets*.
Confirm by clicking *Install AWS* in the popup. 
 
[role="screenshot"]
image::images/firehose-integrations-install-assets.png[AWS integration settings page with the "Install AWS assets" button highlighted]

> We also added an integration called Amazon Kinesis Data Firehose to automatically route AWS service logs to the respective integrations, which include field mappings, ingest pipelines, predefined dashboards and ect.

Reviewer: i think we probably need to revisit this guide in general when we add instructions for the new routing integration. let's leave it out for now, if that's ok?

we can start to draft some better onboarding instructions with the new package in a seperate change.

Author: Will do! Sorry for some reason I missed your comments for 2 weeks 😂

* Elastic recommends setting the *`es_datastream_name`* parameter to help route data to the correct integration data streams.
If this parameter is not specified, data is sent to the *`logs-generic-default`* data stream by default.

[WARNING]
====
The default data stream will change to `logs-awsfirehose-default` in January 2024. *To avoid breaking changes, do not leave `es_datastream_name` empty.*
If you want to try the new https://docs.elastic.co/integrations/awsfirehose[routing functionality], you should explicitly set `es_datastream_name` to `logs-awsfirehose-default`.
====

* You can use the *`es_datastream_name`* parameter to route documents to any data stream.

> When Amazon Kinesis Data Firehose integration is installed, routing will be done automatically with `es_datastream_name` sets to `logs-awsfirehose-default`.

Reviewer 2: ```suggestion
+
When using the https://docs.elastic.co/integrations/awsfirehose[Amazon Kinesis Data Firehose integration], set `es_datastream_name` to `logs-awsfirehose-default` to automatically route your data.
```

Author: These two `when`s are actually both under the same bullet point :)

> When using Elastic AWS integrations without the Firehose integration, you must set this parameter to specific data streams like `logs-aws.vpcflow-default` for ingesting VPC flow logs.

Reviewer 2: ```suggestion
+
When using Elastic AWS integrations without the Firehose integration, you must set `es_datastream_name` to a specific data stream like `logs-aws.vpcflow-default` for ingesting VPC flow logs.
```

* Elastic recommends setting the *`es_datastream_name`* parameter to help route data to the correct integration data streams.
If this parameter is not specified, data is sent to the *`logs-generic-default`* data stream by default.

[WARNING]
====

> The default data stream will change to `logs-awsfirehose-default` in the near future.

Reviewer: Are we able to be more explicit here about timings? I also had a go at making the instructions a little clearer, what do you think? 

> The default data stream will change to `logs-awsfirehose-default` in January 2024. **To avoid breaking changes, do not leave `es_datastream_name` empty.** If you want to try the new [routing functionality](https://docs.elastic.co/integrations/awsfirehose), you should explicitly set `es_datastream_name` to `logs-awsfirehose-default`.
